src/compas_fab/backends/ros/backend_features/move_it_inverse_kinematics.py

```diff
@@ -108,12 +111,22 @@ def inverse_kinematics_async(self, callback, errback,
 
         constraints = convert_constraints_to_rosmsg(options.get('constraints'), header)
 
+        timeout_in_secs = options.get('timeout', 2)
+        timeout_duration = Duration(timeout_in_secs, 0).to_data()
+
         ik_request = PositionIKRequest(group_name=group,
                                        robot_state=start_state,
                                        constraints=constraints,
                                        pose_stamped=pose_stamped,
                                        avoid_collisions=options.get('avoid_collisions', True),
-                                       attempts=options.get('attempts', 8))
+                                       attempts=options.get('attempts', 8),
+                                       timeout=timeout_duration)
+
+        # The field `attempts` was removed in Noetic (and higher)
+        # so it needs to be removed from the message otherwise it causes a serialization error
+        # https://github.com/ros-planning/moveit/pull/1288
+        if self.ros_client.ros_distro not in ('kinetic', 'melodic'):
```

Comment:
It feels like there should be a `Ros_distro` enum.

Reply:
added a minimal one on https://github.com/compas-dev/compas_fab/pull/279/commits/032799c424c5eee23d148235a695571e25835d95